src/textual/widgets/text_input.py

```diff
@@ -40,6 +40,8 @@ def on_key(self, event: events.Key) -> None:
         key = event.key
         if key == "escape":
             return
+        elif key == "space":
```

Comment:
Otherwise when we press the space bar, it would literally write "space" into the text input.